connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -744,6 +746,39 @@ class SparkConnectPlanner(val session: SparkSession) {
     }
   }
 
+  private def transformDeduplicateWithinWatermark(
```

Comment:
Can we reuse `transformDeduplicate()`. We can pass in 'isWithinWatermark' flag to it. This code looks exactly same of that.

Reply:
Sure, will update